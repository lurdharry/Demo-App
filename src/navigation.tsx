import * as React from "react";

import {
  createStackNavigator,
  StackNavigationOptions,
} from "@react-navigation/stack";
import Home from "./screen/home";
import Detail from "./screen/details";
import { Eventdata } from "./screen/types";

export type NavigationParamList = {
  Home: undefined;
  Detail: { event: Eventdata };
};

const options: StackNavigationOptions = {
  headerShown: false,
};

const RootStack = createStackNavigator<NavigationParamList>();

export default function MainNavigator() {
  return (
    <RootStack.Navigator initialRouteName="Home" screenOptions={options}>
      <RootStack.Screen name="Home" component={Home} />
      <RootStack.Screen name="Detail" component={Detail} />
    </RootStack.Navigator>
  );
}
