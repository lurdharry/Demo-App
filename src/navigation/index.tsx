import * as React from "react";
import "react-native-gesture-handler";
import { StackNavigationOptions } from "@react-navigation/stack";
import {
  createSharedElementStackNavigator,
  SharedElementsComponentConfig,
} from "react-navigation-shared-element";
import Home from "../screen/home";
import Detail from "../screen/details";
import { NavigationParamList } from "../types/types.d";

const options: StackNavigationOptions = {
  headerShown: false,
  cardStyleInterpolator: ({ current: { progress } }) => {
    return {
      cardStyle: {
        opacity: progress,
      },
    };
  },
};

const SharedConfig: SharedElementsComponentConfig = route => {
  const data = route.params;
  return [
    {
      id: `item.${data.event.id}.icon`,
      resize: "stretch",
      align: "center-top",
      animation: "fade",
    },
    {
      id: `item.${data.event.id}.title`,
      resize: "stretch",
      align: "left-top",
      animation: "fade",
    },
  ];
};

const RootStack = createSharedElementStackNavigator<NavigationParamList>();

export default function MainNavigator() {
  return (
    <RootStack.Navigator initialRouteName="Home" screenOptions={options}>
      <RootStack.Screen name="Home" component={Home} />
      <RootStack.Screen
        name="Detail"
        component={Detail}
        sharedElementsConfig={SharedConfig}
      />
    </RootStack.Navigator>
  );
}
