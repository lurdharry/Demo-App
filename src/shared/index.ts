export * from "./text";
export * from "./utils";
export * from "./header";
export * from "./card";
