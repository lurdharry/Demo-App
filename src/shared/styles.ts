import { StyleSheet } from "react-native";
import colors from "../constants/colors";
import { wp, hp, paddingTopiOS } from "./layout";

export const ButtonStyles = StyleSheet.create({
  button: {
    backgroundColor: colors.purple,
    width: wp(363),
    alignItems: "center",
    justifyContent: "center",
    height: hp(48),
    alignSelf: "center",
    borderRadius: hp(4),
  },
  title: {
    color: colors.white,
    fontFamily: "roboto-medium",
  },
  greyButton: {
    backgroundColor: colors.grey,

    borderWidth: hp(1),
    borderColor: colors.border,
  },
  title2: {
    color: colors.nobel,
  },
});

export const CardStyles = StyleSheet.create({
  touchItem: {
    flexDirection: "row",
    alignItems: "center",
  },
  checked: {
    opacity: 0.4,
  },
  card: {
    width: wp(363),
    backgroundColor: colors.white,
    alignSelf: "center",
    padding: wp(16),
    borderRadius: hp(8),
    borderWidth: hp(2),
    borderColor: colors.border,
    marginBottom: hp(16),
  },
  left: {
    marginLeft: wp(16),
    alignItems: "flex-start",
  },
  name: {
    fontSize: hp(14),
    color: colors.nobel,
    lineHeight: hp(24),
  },
  date: {
    fontSize: hp(12),
    color: colors.lynch,
    lineHeight: hp(18),
  },
});

export const HeaderStyles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.grey,
    paddingTop: paddingTopiOS,
  },
  header: {
    width: "100%",
    backgroundColor: colors.white,
    paddingVertical: hp(8),
    alignItems: "center",
    justifyContent: "center",
  },
  name: {
    color: colors.lynch,
    fontSize: hp(12),
  },
  event: {
    color: colors.nobel,
    fontSize: hp(16),
  },
  detailBox: {
    paddingBottom: hp(28),
    paddingHorizontal: wp(26),
    backgroundColor: colors.pablue,
    paddingTop: hp(8),
    alignItems: "center",
    width: "100%",
  },
  headerTop: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    marginBottom: hp(35),
    width: "100%",
  },
  nameBox: {
    alignItems: "center",
    justifyContent: "center",
  },
  detailName: {
    color: colors.white,
    fontSize: hp(12),
    lineHeight: hp(18),
  },
  detailEvent: {
    color: colors.white,
  },
  empty: {
    height: hp(24),
    width: wp(24),
  },
});

export const TextStyles = StyleSheet.create({
  RegularText: {
    fontSize: hp(12),
    lineHeight: hp(18),
    color: colors.black,
    fontFamily: "roboto-regular",
  },

  medium: {
    fontSize: hp(12),
    lineHeight: hp(18),
    color: colors.black,
    fontFamily: "roboto-medium",
  },
});
