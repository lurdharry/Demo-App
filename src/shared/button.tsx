import React from "react";
import {
  TouchableOpacityProps,
  ViewStyle,
  TouchableOpacity,
  TextStyle,
  Text,
} from "react-native";

import { ButtonStyles as styles } from "./styles";

interface props extends TouchableOpacityProps {
  title: string;
  style?: ViewStyle;
  titleStyle?: TextStyle;
  isGrey?: boolean;
}

export const Button: React.FC<props> = ({
  style,
  title,
  titleStyle,
  isGrey,
  ...rest
}) => (
  <TouchableOpacity
    style={[styles.button, isGrey && styles.greyButton, style]}
    activeOpacity={0.7}
    {...rest}
  >
    <Text style={[styles.title, isGrey && styles.title2, titleStyle]}>
      {title}
    </Text>
  </TouchableOpacity>
);
