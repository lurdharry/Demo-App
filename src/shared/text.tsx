import React from "react";
import { Text, TextStyle, TextProps } from "react-native";
import { TextStyles as styles } from "./styles";

interface Props extends TextProps {
  title: string;
  style?: TextStyle;
}

const RegularText: React.FC<Props> = ({ title, style, ...rest }) => (
  <Text style={[styles.RegularText, style]} {...rest}>
    {title}
  </Text>
);

const MediumText: React.FC<Props> = ({ title, style, ...rest }) => (
  <Text style={[styles.medium, style]} {...rest}>
    {title}
  </Text>
);

export { RegularText, MediumText };
