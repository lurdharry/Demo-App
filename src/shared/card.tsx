import React, { ReactChild } from "react";
import {
  View,
  ViewStyle,
  TouchableOpacity,
  TouchableOpacityProps,
} from "react-native";
import { formatDate } from "./utils";
import { MediumText, RegularText } from "./text";
import { CardStyles as styles } from "./styles";

interface props extends TouchableOpacityProps {
  name: string;
  date: string;
  checked?: boolean;
  icon: ReactChild;
  style?: ViewStyle;
  onPress: () => void;
}

export const EventCard: React.FC<props> = ({
  name,
  date,
  checked,
  icon,
  style,
  onPress,
}) => {
  const Date = formatDate(date);
  return (
    <TouchableOpacity
      activeOpacity={0.7}
      style={[styles.card, checked && { opacity: 0.5 }, style]}
      onPress={onPress}
    >
      {icon}
      <View style={styles.left}>
        <MediumText title={name} style={styles.name} />
        <RegularText title={Date} style={styles.date} />
      </View>
    </TouchableOpacity>
  );
};
