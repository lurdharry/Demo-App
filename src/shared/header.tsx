import React, { ReactChild } from "react";
import { View, Text } from "react-native";
import { HeaderStyles as styles } from "./styles";
import { TouchableOpacity } from "react-native-gesture-handler";
import { BackArrow, Heart } from "../../assets/svgs";
import { MediumText, RegularText } from "./text";
import colors from "../constants/colors";
import { SharedElement } from "react-navigation-shared-element";

interface props {
  name: string;
}

interface detailProps extends props {
  onPress: () => void;
  icon: ReactChild;
  id: string;
}

export const Header: React.FC<props> = ({ name }) => (
  <View style={styles.header}>
    <SharedElement id="header" style={{ alignItems: "center" }}>
      <RegularText title={name} style={styles.name} />
    </SharedElement>
    <RegularText title="Events" style={styles.event} />
  </View>
);

export const DetailScreenHeader: React.FC<detailProps> = ({
  name,
  onPress,
  icon,
  id,
}) => (
  <View style={styles.detailBox}>
    <View style={styles.headerTop}>
      <TouchableOpacity onPress={onPress} activeOpacity={0.7}>
        <BackArrow />
      </TouchableOpacity>

      <View style={styles.nameBox}>
        <SharedElement id="header">
          <RegularText title={name} style={styles.detailName} />
        </SharedElement>
        <MediumText title="Event" style={styles.detailEvent} />
      </View>

      <View style={styles.empty} />
    </View>
    <SharedElement id={id}>{icon}</SharedElement>
  </View>
);
