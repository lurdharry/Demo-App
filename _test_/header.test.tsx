import React from "react";
import renderer from "react-test-renderer";
import { Text } from "react-native";
import { Header, DetailScreenHeader } from "../src/shared/header";
import { RegularText } from "../src/shared";

jest.mock("react-native-shared-element", () => {
  const { View } = require("react-native");
  const SharedElement = (props: any) => {
    return <View>{props.children}</View>;
  };
  return {
    __esModule: true,
    // default: SharedElement,
    SharedElement: SharedElement,
  };
});

describe("Header", () => {
  it("HeaderBar component renders correctly", () => {
    const text = "Jack Dorsey";
    const instance = renderer.create(<Header name={text} />);
    expect(instance).toMatchSnapshot();
  });
});
