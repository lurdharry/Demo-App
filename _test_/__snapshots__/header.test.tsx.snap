// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header HeaderBar component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "justifyContent": "center",
      "paddingVertical": 14.5,
      "width": "100%",
    }
  }
>
  <View>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "roboto-regular",
            "fontSize": 22,
            "lineHeight": 33,
          },
          Object {
            "color": "#8F92B2",
            "fontSize": 22,
          },
        ]
      }
    >
      Jack Dorsey
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "roboto-regular",
          "fontSize": 22,
          "lineHeight": 33,
        },
        Object {
          "color": "#343866",
          "fontSize": 29,
        },
      ]
    }
  >
    Events
  </Text>
</View>
`;
